lib/integrations/awsoidc/eks_list_clusters.go

```diff
@@ -0,0 +1,186 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package awsoidc
+
+import (
+	"context"
+	"strings"
+	"sync"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/arn"
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	eksTypes "github.com/aws/aws-sdk-go-v2/service/eks/types"
+	"github.com/gravitational/trace"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/gravitational/teleport/api/types"
+)
+
+// ListEKSClustersRequest contains the required fields to list AWS EKS Clusters.
+type ListEKSClustersRequest struct {
+	// Region is the AWS Region.
+	Region string
+
+	// NextToken is the token to be used to fetch the next page.
+	// If empty, the first page is fetched.
+	NextToken string
+}
+
+// CheckAndSetDefaults checks if the required fields are present.
+func (req *ListEKSClustersRequest) CheckAndSetDefaults() error {
+	if req.Region == "" {
+		return trace.BadParameter("region is required")
+	}
+
+	return nil
+}
+
+// EKSCluster represents a cluster in AWS EKS.
+type EKSCluster struct {
+	// Name is the name of AWS EKS cluster.
+	Name string
+
+	// Region is an AWS region.
+	Region string
+
+	// Labels are labels of a EKS cluster.
+	Labels map[string]string
+
+	// ExtraLabels are additional Teleport labels that should be injected into kube agent
+	// if the cluster will be enrolled into Teleport (agent installed on it).
+	ExtraLabels map[string]string
```

Comment:
I don't think `ExtraLabels` indicates what those labels really are.

Should we call them `JoinLabels` or `SuggestedJoinLabels`?

Reply:
Right, changed to `JoinLabels` 👍